### PR TITLE
docs: Document PBS CUDA_VISIBLE_DEVICES requirement [FOUNDENG-359]

### PR DESCRIPTION
**Pull request by jerryharrow**
## Description

PBS should be configured to provide CUDA_VISIBLE_DEVICES. Provide guidance on how to use more than 1 GPU when this is not provided.  Also add some clarifications to directory accessibility in the image caching setup page.

image

image

image

## Checklist
- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by phillip-gaisford — APPROVED** [review]

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst [hunk 1]
```diff
@@ -116,6 +116,14 @@ to optimize how Determined interacts with PBS:
    other configurations such as targeting a specific resource pool with only GPU nodes, or
    specifying a PBS constraint in the experiment configuration.
 
+   PBS should be configured to provide the environment variable ``CUDA_VISIBLE_DEVICES``
+   (``ROCR_VISIBLE_DEVICES`` for ROCm) to the job based upon the GPUs allocated. If PBS is not
+   configured to provide ``CUDA_VISIBLE_DEVICES``, Determined will utilize a single GPU on each
```
Comment (rcorujo) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:121]: I would suggest being a little more explicit as to the mechanism that's being referred to here:

"... configured to set ``CUDA_VISIBLE_DEVICES`` using a PBS cgroup hook as described in the PBS Admin Guide, Determined will ..."

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst [hunk 2]
```diff
@@ -116,6 +116,14 @@ to optimize how Determined interacts with PBS:
...
+   node. To fully utilize mutliple GPUs, you must either manually define ``CUDA_VISIBLE_DEVICES`` or
```
Comment (rcorujo) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:122]: Do we need to say "``CUDA_VISIBLE_DEVICES`` environment variable" to make it clear that it's not a particular setting?

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst [hunk 3]
```diff
@@ -116,6 +116,14 @@ to optimize how Determined interacts with PBS:
...
+   provide the ``pbs.slots_per_node`` setting in your experiment configuration to indicate how many
+   GPU slots are available for use. The format of ``CUDA_VISIBLE_DEVICES`` is an array of GPU IDs
+   (e.g. ``[0,1,2,3]`` which indicates the four GPUS 0 through 3).
```
Comment (rcorujo) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:125]: Perhaps the square brackets are accepted, but I've only see CUDA_VISIBLE_DEVICES be a comma separated list of numbers (e.g., "CUDA_VISIBLE_DEVICES=0,1,2,3").

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst [hunk 4]
```diff
@@ -116,6 +116,14 @@ to optimize how Determined interacts with PBS:
    other configurations such as targeting a specific resource pool with only GPU nodes, or
    specifying a PBS constraint in the experiment configuration.
 
+   PBS should be configured to provide the environment variable ``CUDA_VISIBLE_DEVICES``
+   (``ROCR_VISIBLE_DEVICES`` for ROCm) to the job based upon the GPUs allocated. If PBS is not
+   configured to provide ``CUDA_VISIBLE_DEVICES``, Determined will utilize a single GPU on each
+   node. To fully utilize mutliple GPUs, you must either manually define ``CUDA_VISIBLE_DEVICES`` or
+   provide the ``pbs.slots_per_node`` setting in your experiment configuration to indicate how many
+   GPU slots are available for use. The format of ``CUDA_VISIBLE_DEVICES`` is an array of GPU IDs
```
Comment (rcorujo) [inline docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:124]: Perhaps there should be a sentence, before "The format ...",  that says,

For example, if ``pbs.slots_per_node`` is set to 4, then CUDA_VISIBLE_DEVICES will be automatically set to "0,1,2,3" if it hasn't already been set.

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst (hunk 1 above)
Comment (jerryharrow) [reply docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:121]: ok

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst (hunk 2 above)
Comment (jerryharrow) [reply docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:122]: Since it says 'environment variable' in the very first setting, it should be pretty clear.   Repeating it at each instance seemed to be too wordy.

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst (hunk 3 above)
Comment (jerryharrow) [reply docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:125]: I'll change that.

File: docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst (hunk 4 above)
Comment (jerryharrow) [reply docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:124]: We never set CUDA_VISIBLE_DEVICES.    

I debated covering this at all, because it really is not our business :-), but I was attempting to provide a hint such that they didn't naively set CUDA_VISIBLE_DEVICES=4 and expect it to mean 0,1,2,3.     Perhaps, I'll just delete it.

Comment (rcorujo) [reply docs/cluster-setup-guide/deploy-cluster/sysadmin-deploy-on-slurm/slurm-requirements.rst:124]: Ok, I misunderstood what you were trying to say.
